sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager_base.py

```diff
@@ -25,24 +26,102 @@
     ETAG_KEY,
     FEATURE_FLAG_REFERENCE_KEY,
     FEATURE_FLAG_ID_KEY,
+    ALLOCATION_ID_KEY,
 )
 
 FALLBACK_CLIENT_REFRESH_EXPIRED_INTERVAL = 3600  # 1 hour in seconds
 MINIMAL_CLIENT_REFRESH_INTERVAL = 30  # 30 seconds
 
+JSON = Mapping[str, Any]
+
 
 @dataclass
 class _ConfigurationClientWrapperBase:
     endpoint: str
 
+    @staticmethod
+    def _generate_allocation_id(feature_flag_value: Dict[str, JSON]) -> Optional[str]:
+        """
+        Generates an allocation ID for the specified feature.
+        seed=123abc\ndefault_when_enabled=Control\npercentiles=0,Control,20;20,Test,100\nvariants=Control,standard;Test,special # pylint:disable=line-too-long
+
+        :param  Dict[str, JSON] feature_flag_value: The feature to generate an allocation ID for.
+        :rtype: str
+        :return: The allocation ID.
+        """
+
+        allocation_id = ""
+        allocated_variants = []
+
+        allocation: Optional[JSON] = feature_flag_value.get("allocation")
+
+        if allocation:
+            # Seed
+            allocation_id = f"seed={allocation.get('seed', '')}"
+
+            # DefaultWhenEnabled
+            if "default_when_enabled" in allocation:
+                allocated_variants.append(allocation.get("default_when_enabled"))
+
+            allocation_id += f"\ndefault_when_enabled={allocation.get('default_when_enabled', '')}"
+
+            # Percentile
+            allocation_id += "\npercentiles="
+
+            percentile = allocation.get("percentile")
+
+            if percentile:
+                percentile_allocations = sorted(
+                    (x for x in percentile if x.percentile_from != x.percentile_to),
+                    key=lambda x: x.percentile_from,
+                )
+
+                for percentile_allocation in percentile_allocations:
+                    if percentile_allocation.variant:
+                        allocated_variants.append(percentile_allocation.variant)
+
+                allocation_id += ";".join(
+                    f"{pa.percentile_from},{pa.variant},{pa.percentile_to}" for pa in percentile_allocations
```

Comment:
```suggestion
                    f"{pa.percentile_from},{base64.b64encode(pa.variant).encode('utf-8')).decode('utf-8')},{pa.percentile_to}" for pa in percentile_allocations
```

I don't think my suggestion works with the null case.